### PR TITLE
Update Feature Matrix with icon legend and fix out-of-date fields

### PR DESCRIPTION
**Pull request by awharrison-28**
### Motivation and Context
Resolves https://github.com/microsoft/semantic-kernel/issues/1187

### Description
- added an icon legend
- formatted planning section to align with other tables
- mark ChromaDB as implemented in python SK
- mark Azure Cognitive Search as partially implemented in .NET

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [ ] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: FEATURE_MATRIX.md [hunk 1]
```diff
@@ -38,21 +45,21 @@
...
-| ChromaDb (Memory)                 | ❌ | 🔄 | |
+| ChromaDb (Memory)                 | ❌ | ✅ | |
 | Milvus (Memory)                   | ❌ | ❌ | Vector optimized |
 | Pinecone (Memory)                 | ✅ | ❌ | Vector optimized |
 | Weaviate (Memory)                 | ❌ | ❌ | Vector optimized |
 | CosmosDB (Memory)                 | ✅ | ❌ | CosmosDB is not optimized for vector storage |
 | Sqlite (Memory)                   | ✅ | ❌ | Sqlite is not optimized for vector storage |
 | Postgres (Memory)                 | ✅ | ❌ | Vector optimized (required the [pgvector](https://github.com/pgvector/pgvector) extension) |
-| Azure Cognitive Search            | ❌ | ❌ | |
+| Azure Cognitive Search            | 🔄 | ❌ | Azure Search backed by vector search coming soon |
```
Comment (lemillermicrosoft) [inline FEATURE_MATRIX.md:62]: I'd mark as enabled for now and allow any PRs that are doing changes in this space to update as appropriate.

File: FEATURE_MATRIX.md [hunk 2]
```diff
@@ -38,21 +45,21 @@
...
-| SequentialPlanner | ✅ | ❌ | |
+| Plan                              | ✅ | ❌ | |
+| SequentialPlanner                 | ✅ | ❌ | |
 
 ## Connectors and Skill Libraries
 | | C# | Python | Notes |
 |---|---|---|---|
 | Qdrant (Memory)                   | ✅ | ❌ | Vector optimized |
-| ChromaDb (Memory)                 | ❌ | 🔄 | |
+| ChromaDb (Memory)                 | ❌ | ✅ | |
```
Comment (lemillermicrosoft) [inline FEATURE_MATRIX.md:55]: Is ChromaDB Vector Optimized? Would it be better to highlight which are not vector optimized?

**Review by lemillermicrosoft — APPROVED** [review]

File: FEATURE_MATRIX.md (hunk 2 above)
Comment (shawncal) [reply FEATURE_MATRIX.md]: Chroma is an embedding db, so I believe it must be vector optimized. Per your suggestion, I've removed the "Vector optimized" on all of the entries, only calling out the ones that are not.

**Review by shawncal — APPROVED** [review]

**Comment by 1983Thai** [comment]
b4d555d6552076a7f581481a1905578e989bffa5

**Pull request merged**